src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -168,16 +130,38 @@ private void OnOutputDataReceived(object sender, DataReceivedEventArgs e)
         {
             if (e.Data != null)
             {
-                string msg = e.Data;
-                if (WorkerProcessUtilities.IsConsoleLog(msg))
-                {
-                    msg = WorkerProcessUtilities.RemoveLogPrefix(msg);
-                    _workerProcessLogger?.LogInformation(msg);
-                }
-                else
-                {
-                    _consoleLogSource?.Log(msg);
-                }
+                BuildAndLogConsoleLog(e.Data, LogLevel.Information);
+            }
+        }
+
+        private void BuildAndLogConsoleLog(string msg, LogLevel level = LogLevel.Information)
+        {
+            ConsoleLog consoleLog = new ConsoleLog()
+            {
+                Message = msg,
+                Level = level
+            };
+
+            if (msg.IndexOf("warn", StringComparison.OrdinalIgnoreCase) > -1)
+            {
+                consoleLog.Level = LogLevel.Warning;
+            }
+            else if (_useStdErrStreamForErrorsOnly || (msg.IndexOf("error", StringComparison.OrdinalIgnoreCase) > -1) ||
```

Comment:
`else if (_useStdErrStreamForErrorsOnly)` - wondering if this works as intended when stdout and stderr and use this method? It seems like if `_useStdErrStreamForErrorsOnly` is true, everything gets logged as an error, no matter what?

Reply:
thank you. I over-did refactoring. Updated and also tested following:

```js
console.error("eTest msg r");  // Error log
console.info("Test info"); // Info log
console.warn("w:Test magg2"); // error log
console.warn("warning:Test magg2"); // warning as this is prefixed with warning
```